aws-sdk-resources/features/s3/download_file.feature

```diff
@@ -0,0 +1,57 @@
+# language: en
+@s3 @resources @downloadfile
+Feature: Managed file download
+
+ Background:
+    Given I create a bucket
+
+  @get
+  Scenario: Download a small object
+    Given I have a 1M file
+    And I upload the file using put_object
+    When I download the file with mode "auto"
+    And 1 get_object request should have been made
+    And the downloaded file should match the uploaded file
+
+  # Any object uploaded in parts, is downloaded in parts
+  @get-part
+  Scenario: Download a large object using part numbers
+    Given I have a 16M file
+    And I upload the file
+    When I download the file with mode "auto"
+    Then 4 upload_part requests should have been made
+    And 4 get_object requests should have been made
+    And the downloaded file should match the uploaded file
+
+  # "larger" objects (exceeding 15MB) are downloaded in chunks,
+  # using RANGE requests, even if the it was not uploaded in parts
+  @get-range
+  Scenario: Download a large object using ranges
+    Given I have a 16M file
+    And I upload the file using put_object
+    When I download the file with mode "auto"
+    Then 4 get_object requests should have been made
+    And the downloaded file should match the uploaded file
```

Comment:
Could we also integration test our cleanup of temp files? That would be a bad regression if it ever happened.